content/cpp/concepts/user-input/terms/getline/getline.md

```diff
@@ -0,0 +1,76 @@
+---
+Title: 'getline()'
+Description: 'Converts user input into a string delimited by a character, if specified, and then stores it into a variable.'
+Subjects:
+  - 'Computer Science'
+  - 'Game Development'
+Tags:
+  - 'Functions'
+  - 'Strings'
+  - 'Characters'
+  - 'Data Types'
+CatalogContent:
+  - 'learn-c-plus-plus'
+  - 'paths/computer-science'
+---
+
+In C++, the `getline()` function converts user input into a character-delimited string and stores them in a variable. If a delimiting character is not specified, then the entire input will be stored.
+
+## Syntax
+
+The `getline()` function is part of the `<string>` header.
+
+```pseudo
+#include <string>
+
+getline(cin, string, delim)
+```
+
+- `getline()` is part of the `<string>` header, so it is included at the top of the file.
+- `cin` is the object in the stream, which would be the user input.
+- `string` refers to the variable the user input is set to.
+- `delim` refers to a character that the user's input is stored up to.
+
+Although not recommended, `getline()` can be called on the `cin` object as a member function like this:
```

Comment:
```suggestion
Although not recommended in C++, since it deals with C strings, `getline()` can be called on the `cin` object as a member function like this:
```

Reply:
I'm going to move this to C, though do you think this should go into user input there as well? or strings 🤔 @Dusch4593 @KTom101

Reply:
on second thought, I'm going to leave this in C++ because I see that `cin` input stream object does not exist in C. For `getline`, it looks like the reason `cin.getline()` is not recommended is that it works with character arrays instead of C++ strings.